### PR TITLE
[WIP] Switch to std futures

### PR DESCRIPTION
**Pull request by djc**
Open issues:

* For some reason the `poll()` implementation cannot be found in a few places?
* mio doesn't seem to have a `std-future` branch yet

### CODE REVIEW

**Comment by Ralith** [comment]
> For some reason the `poll()` implementation cannot be found in a few places?

`Pin::new(&mut foo).poll(cx)`

> mio doesn't seem to have a `std-future` branch yet

What does tokio std-futures use?

File: quinn/src/streams.rs [hunk 1]
```diff
@@ -374,20 +381,17 @@ impl From<ReadError> for ReadToEndError {
...
-
 impl AsyncRead for RecvStream {
-    unsafe fn prepare_uninitialized_buffer(&self, _: &mut [u8]) -> bool {
-        false
+    fn poll_read(
+        self: Pin<&mut Self>,
+        cx: &mut Context,
+        buf: &mut [u8],
+    ) -> Poll<Result<usize, io::Error>> {
+        match self.read(cx, buf) {
```
[the remark this answers is not in the document]
Comment (djc) [reply quinn/src/streams.rs]: We can probably use the `?` operator instead, right?

**Comment by djc** [comment]
> I actually already wrote much of this, only stopping at the tests (with the lib compiling) when I realized I wasn't sure if futures 0.3 would be released at the same time as std futures, since it has lots of non-'static futures that seem awkward to use before async/await, e.g. its ReadToEnd variant. I've pushed a std-futures branch (which I now realize is a very similar name, oops) for comparison. My apologies for the poor communication.

Ahh, I was wondering if you'd done the work already. I figured I'd open a WIP PR proactively to let you know about the work. So I think it's good if you own this topic for now (maybe open a PR with your take, or take over this branch?), and I'll focus on some of the other stuff?

> I'm not sure I like having our inherent poll_read/write methods renamed to read/write; the poll_ names are more correct, UFCS should be sufficient to disambiguate when necessary, and the normal behavior, preferring inherent impls, is desired here.

Yeah, I think I agree.

**Comment by Ralith** [comment]
> So I think it's good if you own this topic for now (maybe open a PR with your take, or take over this branch?), and I'll focus on some of the other stuff?

Works for me, closing in favor of #372.

**Pull request closed without merging**